pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -153,42 +153,21 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
 			newCRB := newClusterRoleBinding(clusterRoleBindingName)
 			err := r.client.Create(context.TODO(), newCRB)
 			if err != nil {
-				groupPermissionConditions := instance.Status.Conditions
-
-				// make new condition
-				newCondition := managedv1alpha1.Condition{
-					LastTransitionTime: metav1.Now(),
-					ClusterRoleName:    clusterRoleName,
-					Message:            "Unable to create ClusterRoleBinding: <error from platform>",
-					Status:             true,
-					State:              "Failed",
-				}
-
-				// update GroupPermission object with new conditions
-				instance.Status.Conditions = append(groupPermissionConditions, newCondition)
+				// calls on helper function to update the condition of the groupPermission object
+				groupPermissionObject := updateCondition(instance, "Unable to create ClusterRoleBinding: <error from platform>", clusterRoleName, true, "Failed")
```

Comment:
Why are you assigning to a new object instead of reassigning to `instance`>

Reply:
Where does `<error from platform>` come from?

Reply:
error from platform is err, will pass that in as a string